packages/client/src/index.js

```diff
@@ -254,16 +252,34 @@ export let ContextTree = _.curry(
 
     let actionProps = {
       getNode,
-      flat,
       dispatch,
-      snapshot,
       extend,
+      snapshot,
       types,
       initNode,
       initObject,
+      log,
     }
+    F.extendOn(
+      actionProps,
+      _.pick(
```

Comment:
I think it's probably safe to just include all actions, but we can revisit that after dropping the flat tree